core/domain/rule_domain_test.py

```diff
@@ -98,10 +100,47 @@ def test_that_all_rules_have_object_editor_templates(self):
                     self.assertTrue(
                         param_obj_type.has_editor_js_template(),
                         msg='(%s)' % clazz.description)
-                    at_least_one_rule_found = True
-                clses.append(clazz)
 
-        self.assertTrue(at_least_one_rule_found)
+        self.assertEqual(EXPECTED_TOTAL_NUMBER_OF_RULES, num_rules)
```

Comment:
Hmm, I'm not very sure about the pattern of having an expected rule count. It doesn't seem very clear to me that a developer will know to increase this count, unless it is better documented. (Otherwise, if people forget to increment this, the expected rule count doesn't really serve its purpose.)

What was the check that num_rules is not zero supposed to do? (Why do we want to know that there is at least one rule?)

Also, why is the expected rules check repeated in the test below?